src/doc.rs

```diff
@@ -23,6 +23,7 @@ Additional arguments:
 --module path   - path to executable or dynamic library from which to load debug symbols; useful if the running executable is stripped and you have an unstripped version on the side; can be specified multiple times to provide multiple dynamic libraries (they'll be automatically matched by build id)
 -o   - try to get debug info from debuginfod server at https://debuginfod.elfutils.org/ ; alternatively, set environment variable DEBUGINFOD_URLS to a space-separated list of URLs to use
 --mouse full|no-hover|disabled   - mouse mode; 'no-hover' to react only to clicking and dragging, 'disabled' to disable mouse altogether; default is 'full' (if it doesn't work, check if mouse reporting is enabled in the terminal application)
+--breakpoint path:line   - set a breakpoint at the specified source file and line number (e.g. src/main.c:42); repeat the parameter to set multiple breakpoints
```

Comment:
Just realized this will probably be annoying to use because the path has to exactly match what's in debug symbols, which may be a path relative to some build directory and may include nonsense like `./build/../src/main.c`. Let's at least add a warning:
```suggestion
--breakpoint path:line   - set a breakpoint at the specified source file and line number (e.g. src/main.c:42); repeat the parameter to set multiple breakpoints; the path must exactly match a path appearing in debug info
```

If anyone uses this feature we should probably (in a separate PR) make it match paths properly by having a flag in LineBreakpoint saying that the path is not exact, then doing a search in Symbols later, probably in `determine_locations_for_breakpoint`. Then e.g. `src/main.c` would be enough instead of `/home/someone/foo/bar/src/main.c`.

Reply:
Yes agree - some sort of matching of the paths would be nice.

Reply:
If you're up for doing it, I *think* it's as easy as adding a search in determine_locations_for_breakpoint - iterate over the array of all filenames in debug info (Symbols.file_paths), do some kind of heuristic matching between two paths (outputting a score, then picking the match with highest score; e.g. how many last path components match, after some kind of normalization to remove '..').

Reply:
Sure - I'm up for giving it a shot :)

Reply:
It seems like I can't just add some search functionality to `determine_locations_for_breakpoint` because while the debugger breaks - viewing the breakpoint, results in it not being able to find the file.

![image](https://github.com/user-attachments/assets/08d27b13-e603-4878-a3d9-a33cbcf492a2)

If we correct the path somehow, then we also have to look at the duplication problem again.

```rust
for line_breakpoint in &context.settings.breakpoints {
    let existing_id = debugger.find_line_breakpoint_fuzzy(&line_breakpoint);
    if let Some(id) = existing_id {
        let _ = debugger.set_breakpoint_enabled(id, true);
    } else {
        let _ = debugger.add_breakpoint(BreakpointOn::Line(line_breakpoint.clone()));
    }
}
```

Maybe adding the search logic to `add_breakpoint` - but only when there's not a direct match. Then have LineBreakpoint store both the path and the orignal path - where the original path can be used for the deduplication?

Reply:
> viewing the breakpoint, results in it not being able to find the file

`determine_locations_for_breakpoint` should reassign `LineBreakpoint.path` with the path from symbols.

> then we also have to look at the duplication problem again

Ah, right, this breaks deduplication. Storing the original path just for deduplication would work. Alternatively, `determine_locations_for_breakpoint` can just outright delete the breakpoint if it turns out to be duplicate after updating its `path` (and activate the other breakpoint if the deleted one was active) (all of this only if the breakpoint is explicitly marked as needing lookup, and this mark should be cleared after lookup is done once). It'll look a little weird in the UI: the breakpoint will disappear from the list when symbols are loaded; but that seems ok. Pick whichever solution seems better, I can't think of strong reasons to prefer one or the other.

Make sure to add all the new breakpoint fields to `Breakpoint.save_state/load_state`.

The UI should probably avoid auto-opening breakpoint's file before its path is resolved. Just add a check next to the `state.should_scroll_source` assignment in `BreakpointsWindow`.

(There'll be another small UI quirk: if `determine_locations_for_breakpoint` updates breakpoint's path after the user selected this breakpoint in the `breakpoints` window, the `code` window won't automatically jump to the newly resolved path (like it jumps when moving the cursor in breakpoints window). We could make it do that, but that would be even worse: maybe the user wasn't looking at the breakpoints window and was doing something else in the code window. I guess we shouldn't do anything, other than the previous paragraph. There was a similar problem with stack trace, and I did something complicated with it which is still janky and maybe wasn't worth it.)